Sources/SwifterSwift/SwiftStdlib/OptionalExtensions.swift

```diff
@@ -111,6 +111,40 @@ public extension Optional where Wrapped: Collection {
 
 }
 
+// MARK: - Methods (BinaryFloatingPoint)
+public extension Optional where Wrapped: BinaryFloatingPoint {
+    
+    /// SwifterSwift: Check if optional is nil or zero float.
+    var isNilOrZero: Bool {
+        guard let float = self else { return true }
+        return float.isZero
+    }
+    
+    /// SwifterSwift: Returns the float only if it is not nill and not zero.
+    var nonZero: Wrapped? {
+        guard let float = self else { return nil }
+        guard !float.isZero else { return nil }
+        return float
+    }
+}
+
+// MARK: - Methods (BinaryInteger)
+public extension Optional where Wrapped: BinaryInteger {
+    
+    /// SwifterSwift: Check if optional is nil or zero integer.
+    var isNilOrZero: Bool {
+        guard let integer = self else { return true }
+        return integer == 0
+    }
+    
+    /// SwifterSwift: Returns the integer only if it is not nill and not zero.
+    var nonZero: Wrapped? {
```

Comment:
The `isNilOrZero` fine but, can you provide some use cases for this one? I imagine it can be used for `if let i = optionalInt.nonZero { }` but I think `if let int = optionalInt, !int.isZero { }` serve the same purpose and is more clear, so I'm not sure about this. What you think?

Reply:
Sorry, I may not use swiftlint autocorrection to eliminate these space warnings.😅
I think`isNilOrZero` still has to exist, just like`isNilOrEmpty` is on`String`
I don't know why I have to add a file to add the`isZero` property to`BinaryIntegerExtensions`, there will always be errors about `Codecov changes.`
I tried to solve it in the past, but it didn't work.
I am sorry that I am not good at this aspect of the pr code, Can you help me?

Reply:
Agree.  Equating optionality and a magic value seems very un-Swifty.  If you want to perform the same action when a value is missing or when it is zero, that seems more like business logic than a general purpose extension to me.

Reply:
@chulbert 
You have some facts that the judgment of 0 and null does have more to do with business logic.😄

Reply:
> I think`isNilOrZero` still has to exist, just like`isNilOrEmpty` is on`String`

I agree, I think is fine, the one I was talking is `nonZero` one :)) 



> I don't know why I have to add a file to add the`isZero` property to`BinaryIntegerExtensions`, there will always be errors about `Codecov changes.`

I don't think you have to add another file, can you provide more info about the error? 


> I am sorry that I am not good at this aspect of the pr code, Can you help me?

Sure :)) Just let us know what the error was and we can figure out together

Reply:
I think`isZero` is added for the`BinaryInteger` protocol. When I looked at the original file structure, I didn't find a file about`BinaryIntegerExtensions`. If I don't add a new file, I should put it there.

Reply:
@LucianoPAlmeida 
It seems that`isNilOrZero` and`nonZero` are really more biased towards the business module, I think that`swifterswift` has `Optional where Wrapped == Collection` to expand the`isNilOrEmpty` and`nonEmpty` properties, then you should need to add`isNilOrZero` and`nonZero` for `Optional where Wrapped == BinaryFloatingPoint` Attributes, in fact, the discussion of`isNilOrEmpty`,`nonEmpty`,`isNilOrZero``nonZero` and are more biased towards business logic. What do you think?

Reply:
Sorry, just note that I've missed a lot of discussions lately here 😅, but if you still like to continue from here.
>  Attributes, in fact, the discussion of`isNilOrEmpty`,`nonEmpty`,`` isNilOrZero``nonZero `` and are more biased towards business logic. What do you think?

I just think that maybe those are over abstracting very simple operations, although some people may think that is more readable and more business logic as you said. I'm not sure is a good fit for here, since we try to gather general propose extension that may be useful in a general way.

Also, if I miss the notification here on GitHub for to much time(like a week or two), feel free to ping me on the SwifterSwift slack :))